server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchDataLoader.java

```diff
@@ -0,0 +1,218 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.qualityprofile.ws;
+
+import com.google.common.base.Predicate;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.builder.CompareToBuilder;
+import org.sonar.api.resources.Language;
+import org.sonar.api.resources.Languages;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.qualityprofile.QualityProfileDto;
+import org.sonar.server.qualityprofile.QProfile;
+import org.sonar.server.qualityprofile.QProfileFactory;
+import org.sonar.server.qualityprofile.QProfileLoader;
+import org.sonar.server.qualityprofile.QProfileLookup;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.FluentIterable.from;
+import static java.lang.String.format;
+import static org.sonar.server.qualityprofile.ws.SearchAction.PARAM_DEFAULT;
+import static org.sonar.server.qualityprofile.ws.SearchAction.PARAM_LANGUAGE;
+import static org.sonar.server.qualityprofile.ws.SearchAction.PARAM_PROFILE_NAME;
+import static org.sonar.server.qualityprofile.ws.SearchAction.PARAM_PROJECT_KEY;
+import static org.sonar.server.ws.WsUtils.checkRequest;
+
+public class SearchDataLoader {
+  private final Languages languages;
+  private final QProfileLookup profileLookup;
+  private final QProfileLoader profileLoader;
+  private final QProfileFactory profileFactory;
+  private final DbClient dbClient;
+
+  public SearchDataLoader(Languages languages, QProfileLookup profileLookup, QProfileLoader profileLoader, QProfileFactory profileFactory, DbClient dbClient) {
+    this.languages = languages;
+    this.profileLookup = profileLookup;
+    this.profileLoader = profileLoader;
+    this.profileFactory = profileFactory;
+    this.dbClient = dbClient;
+  }
+
+  SearchData load(Request request) {
+    validateRequest(request);
+
+    SearchData data = new SearchData()
+      .setProfiles(findProfiles(request))
+      .setActiveRuleCountByProfileKey(profileLoader.countAllActiveRules())
+      .setProjectCountByProfileKey(dbClient.qualityProfileDao().countProjectsByProfileKey())
+      .setFieldsToReturn(request.paramAsStrings(WebService.Param.FIELDS));
+
+    return data;
+  }
+
+  private List<QProfile> findProfiles(Request request) {
+    List<QProfile> profiles;
+    if (askDefaultProfiles(request)) {
+      profiles = findDefaultProfiles();
+    } else if (hasComponentKey(request)) {
+      profiles = findProjectProfiles(request);
+    } else {
+      profiles = findAllProfiles(request);
+    }
+
+    List<QProfile> orderedProfiles = orderProfiles(profiles);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "orderedProfiles". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)